sqlglot/dialects/postgres.py

```diff
@@ -26,6 +28,9 @@
     "DAY": " / 86400",
 }
 
+# This pattern is used to capture valid postgres string intervals like '  15  days '
+INTERVAL_STRING_RE = re.compile("([0-9]+)([a-zA-Z]+)")
```

Comment:
it seems like this could be a generic method, maybe make this a function so it's not only used in generate_series?

also maybe it should be in dialects.py

Reply:
You mean just extracting a number and a var from a string the matches the re? Or creating an Interval expression too?

Reply:
we probably always want an interval right?

Reply:
I guess